### ISSUE TITLE
ccall test failure on arm

### ISSUE THREAD

**Issue by ViralBShah** [JuliaLang/julia#12343]
The ccall test segfaults currently on arm.
````
(precise)viral@localhost:~/julia/test$ ../julia 
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "help()" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.0-dev+6253 (2015-07-26 15:40 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit a97b24b* (1 day old master)
|__/                   |  arm-linux-gnueabihf
````

**Comment by yuyichao** [comment]
Backtrace? Or are you fixing it now?

**Comment by tkelman** [comment]
Proper bug reports, versioninfo (esp LLVM? presumably the default arm setting from Make.inc?), traces, etc would be good for these. Otherwise not much point in having separate bugs for them as opposed to one checklist of all failures.

**Comment by ViralBShah** [comment]
All my tests are using the default llvm in Make.inc. 

The tests are still running (on and off). Once done, I will try to get backtraces. The reason to open this is that someone else may also be able to get the backtrace. There is no point having a checklist of failures in one issue, since discussion gets mixed up - as was seen in the earlier arm testing issue.

**Comment by vtjnash** [comment]
this is caused by using the wrong ABI (it's also why you get a warning when compiling julia codegen that this test won't work)

**Comment by pao** [comment]
cf https://github.com/JuliaLang/julia/issues/10791#issuecomment-91735439; granted nothing else works on ARMv8 right now

**Comment by yuyichao** [comment]
Should be fixed by #14194 now.
